### DISCUSSION TITLE
CRSF_FRAMETYPE_SUBSET_RC_CHANNELS_PACKED number of channels

### DISCUSSION THREAD

**Discussion by mmosca** [crsf-wg/crsf#17] [Q&A]
Are channels supposed to be use all bits available?

Eg.: 11 bits and 13bits needs to be multiple of 8 channels and 12 bits in multiples of 2?

**Comment by kobelev-dev** [comment]
Hi @mmosca, Frankly speaking I am not getting what are you talking about. Can you please explain that in more details?

My reading (not 100% sure that it is correct, nevertheless) of doc on CRSF_FRAMETYPE_SUBSET_RC_CHANNELS_PACKED suggests that some number of bits in the last byte of the payload might be unused.

**Reply by mmosca** [reply]
Where is that written? Are you saying it based on BF's implementation of the crsf parsing example?

It makes sense to have some padding at the end, if you have some unused bits in the last byte. But the alignment requirement is not explicit.

**Reply by kobelev-dev** [reply]
Well, this is not written anywhere directly.

But, if you will look at how the CRSF_FRAMETYPE_RC_CHANNELS_PACKED is implemented, you will see that the data is packed there bit to bit without any padding/aligning. The second point is that the names of the commands are very similar. They differ only in the "_SUBSET" part. This suggests that their implementations are similar too. Different people with different ideas more likely will select a different name.

It would definitely make sense to write that (if my reading between the line is correct) in the spec directly.
